File: lab2/morse.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strings"
)

func main() {
	morse := []string{".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---",
		"-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-",
		"..-", "...-", ".--", "-..-", "-.--", "--.."}

	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Введите слова: ")
	s, _ := reader.ReadString('\n')
	s = strings.TrimSpace(s)

	words := strings.Fields(s)
	unique := make(map[string]bool)

	for _, w := range words {
		chars := strings.Split(w, "")
		sort.Strings(chars)
		sortedWord := strings.Join(chars, "")
		code := ""
		for _, ch := range sortedWord {
			code += morse[ch-'a']
		}
		unique[code] = true
	}

	fmt.Println(len(unique))
}

File: lab2/morse.js
const readline = require('readline');

const morse = [
    ".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---",
    "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-",
    "..-", "...-", ".--", "-..-", "-.--", "--.."
];

function wordToMorse(word) {
    let result = "";
    for (let i = 0; i < word.length; i++) {
        const idx = word.charCodeAt(i) - 97; // 'a' = 97
        result += morse[idx];
    }
    return result;
}

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Введите слова: ", (input) => {
    const words = input.trim().split(/\s+/);
    const uniqueMorse = new Set();
    
    for (let word of words) {
        const sortedWord = word.split('').sort().join('');
        uniqueMorse.add(wordToMorse(sortedWord));
    }
    
    console.log(`Количество: ${uniqueMorse.size}`);
    rl.close();
});
